### PR TITLE
react-redux interop + redux-persist compat + Symbol

### PR DESCRIPTION
**Pull request by takion**
see issue https://github.com/ctrlplusb/easy-peasy/issues/53

### CODE REVIEW

File: src/easy-peasy.js [hunk 1]
```diff
@@ -4,15 +4,25 @@ import {
...
+import produce, { setAutoFreeze } from 'immer'
 import thunk from 'redux-thunk'
 import { isStateObject } from './lib'
 
-const effectSymbol = '__effect__'
-const selectSymbol = '__select__'
-const selectDependenciesSymbol = '__selectDependencies__'
-const selectStateSymbol = '__selectState__'
-const reducerSymbol = '__reducer__'
+setAutoFreeze(false)
```
Comment (ctrlplusb) [inline src/easy-peasy.js]: Interesting. I had no idea about this. Perhaps we should tie the value to the dev boolean config of the store?

File: src/easy-peasy.js [hunk 2]
```diff
@@ -4,15 +4,25 @@ import {
...
+
+const effectSymbol = Symbol('__effect__')
```
Comment (ctrlplusb) [inline src/easy-peasy.js]: We had issues with Symbols - see #16

File: src/easy-peasy.js [hunk 3]
```diff
@@ -4,15 +4,25 @@ import {
...
+const selectSymbol = Symbol('__select__')
+const selectDependenciesSymbol = Symbol('__selectDependencies__')
+const selectStateSymbol = Symbol('__selectState__')
+
+class reducerContainer {
```
Comment (ctrlplusb) [inline src/easy-peasy.js]: Whats the thinking behind having this container?

File: src/easy-peasy.js (hunk 1 above)
Comment (takion) [reply src/easy-peasy.js]: I agree, that would be better

File: src/easy-peasy.js (hunk 2 above)
Comment (takion) [reply src/easy-peasy.js]: Ah ok, maybe you could take the same approach that I took with reducerContainer, a wrapper and then instanceof, to let clean the original references. I think adding property, and especially this way, to parameters will polute them.

File: src/easy-peasy.js (hunk 3 above)
Comment (takion) [reply src/easy-peasy.js]: Compatibility with redux-persist that use Object.freeze api and then lead to an error whan easy-peasy try to modify it, even with adding a Symbol

**Comment by ctrlplusb** [comment]
Closing this as won't merge. But did take your hint at the object freeze disable. Thanks! ❤️

**Pull request closed without merging**